prov/rxd/src/rxd_ep.c

```diff
@@ -1087,6 +1092,9 @@ static void rxd_ep_progress(struct util_ep *util_ep)
 		rxd_progress_pkt_list(ep, &peer->unacked);
 		if (dlist_empty(&peer->unacked))
 			rxd_progress_tx_list(ep, peer);
+		if (dlist_empty(&peer->unacked))
+			rxd_ep_send_ack(ep, rxd_get_peer_idx(ep, peer));
+
```

Comment:
Hmm, yes if it doesn't have any pending outgoing transfers. I think it would need some extra check to only send the ack if we haven't gotten anything from that peer either to avoid that stream. But we still run into the issue of sending miscellaneous acks to peers that aren't currently sending anything while we're waiting for completions from one peer. I'm not sure how to fix this issue. Any ideas?

Reply:
I don't understand the issue that this is trying to solve.  We don't want peers sending acks to each other just in case the peer might need to do something.  At scale, this will  potentially generate a glob (that's a technical term) of acks from each peer to all other peers.  We should just let local timers result in retries and not receiving messages from peers.

Reply:
The issue I was seeing is that when a peer sends a packet and it needs to be resent, if the app is using sread, it's not going to call progress and resend the packet until the cq is signaled by a completion on the dg CQ. So the tx side is waiting on that fd until something comes in.